internal/orchestrator/commiter.go

```diff
@@ -42,13 +45,203 @@ func (c *Commiter) Start() {
 	go func() {
 		for t := range ticker.C {
 			fmt.Println("Commiter running at", t)
-			// TODO: fetch max block number from main table
-			// TODO: fetch sequential block numbers from staging table
-			// TODO: save to main table
-			// TODO: delete from staging table
+			blocksToCommit, err := c.getSequentialBlocksToCommit()
+			if err != nil {
+				log.Printf("Error getting blocks to commit: %v", err)
+				continue
+			}
+			if len(blocksToCommit) == 0 {
+				log.Println("No blocks to commit")
+				continue
+			}
+			if err := c.commit(blocksToCommit); err != nil {
+				log.Printf("Error committing blocks: %v", err)
+			}
 		}
 	}()
 
 	// Keep the program running (otherwise it will exit)
 	select {}
 }
+
+func (c *Commiter) getBlockNumbersToCommit() ([]uint64, error) {
+	maxBlockNumber, err := c.storage.DBMainStorage.GetMaxBlockNumber()
+	if err != nil {
+		return nil, err
+	}
+	startBlock := maxBlockNumber + 1
+	endBlock := maxBlockNumber + uint64(c.blocksPerCommit)
+	var blockNumbers []uint64
+	for i := startBlock; i <= endBlock; i++ {
+		blockNumbers = append(blockNumbers, i)
+	}
+	return blockNumbers, nil
+}
+
+func (c *Commiter) getSequentialBlocksToCommit() ([]common.Block, error) {
+	blocksToCommit, err := c.getBlockNumbersToCommit()
+	if err != nil {
+		return nil, fmt.Errorf("error determining blocks to commit: %v", err)
+	}
+	blocks, err := c.storage.DBStagingStorage.GetBlocks(storage.QueryFilter{BlockNumbers: blocksToCommit})
+	if err != nil {
+		return nil, fmt.Errorf("error fetching blocks to commit: %v", err)
+	}
+	if len(blocks) == 0 {
+		return nil, nil
+	}
+
+	// Sort blocks by block number
+	sort.Slice(blocks, func(i, j int) bool {
+		return blocks[i].Number < blocks[j].Number
+	})
+
+	var sequentialBlocks []common.Block
+	expectedBlockNumber := blocks[0].Number
+
+	for _, block := range blocks {
+		if block.Number != expectedBlockNumber {
+			// Gap detected, stop here
+			break
+		}
```

Comment:
shouldn't this return an error? and what are the cases of gaps here?

Reply:
no, commiter should only insert sequantially. If it can't then the whole commiter shouldn't stop trying, but wait until the next block becomes available